### PR TITLE
[BugFix] Support rewrite_once when the number of callbacks > 1

### PR DESCRIPTION
**Pull request by sisleyli**
The `rewrite_once` of the current `PatternRewriter`  rewrites once only when the last callback in the callbacks list is `rewrite_once`. I don't think this is as expected.   
I think the `rewrite_once` strategy should be that the callback marked with `rewrite_once` is rewritten only once, and the rest of the callbacks are still rewritten multiple times.

### CODE REVIEW

**Comment by sisleyli** [comment]
What do you think @masahi ? Thank you!

File: src/relay/ir/dataflow_matcher.cc [hunk 1]
```diff
@@ -796,24 +796,38 @@ Expr PatternRewriter::Rewrite(const Array<DFPatternCallback>& callbacks, const E
   bool equal = true;
   static auto* structural_equal = runtime::Registry::Get("node.StructuralEqual");
   ICHECK(structural_equal) << "node.StructuralEqual is not registered.";
+  // Use the callback until the callback's attribute is rewrite_once=true and has been rewritten
+  std::unordered_map<DFPatternCallback, bool, ObjectPtrHash, ObjectPtrEqual> callbacks_map;
```
Comment (masahi) [inline src/relay/ir/dataflow_matcher.cc:800]: `callbacks_map` -> `done`, swapping false and true

File: src/relay/ir/dataflow_matcher.cc [hunk 2]
```diff
@@ -796,24 +796,38 @@ Expr PatternRewriter::Rewrite(const Array<DFPatternCallback>& callbacks, const E
   bool equal = true;
   static auto* structural_equal = runtime::Registry::Get("node.StructuralEqual");
   ICHECK(structural_equal) << "node.StructuralEqual is not registered.";
+  // Use the callback until the callback's attribute is rewrite_once=true and has been rewritten
```
Comment (masahi) [inline src/relay/ir/dataflow_matcher.cc:799]: // Keep track of callbacks that have finished rewriting

File: src/relay/ir/dataflow_matcher.cc (hunk 1 above)
Comment (sisleyli) [reply src/relay/ir/dataflow_matcher.cc]: Please take another look. Thank you!

**Review by masahi — APPROVED** [review]

**Comment by sisleyli** [comment]
> It seems that CI has passed, could you please help me merge this PR? Thanks @masahi !

I don't know whether our repo will automatically rebase when it merges PR, so I just performed a rebase operation myself, do I need to do this every time the main branch is updated?

**Comment by masahi** [comment]
No you don't have to do that. You now have to wait for another 4h :)

**Comment by sisleyli** [comment]
Thanks masahi, I should have consulted before performing the rebase operation :(

**Comment by sisleyli** [comment]
CI passed again...I won't do redundant operations this time.hhhh

**Pull request merged**